src/world/terrain/material.hpp

```diff
@@ -65,6 +65,22 @@ struct Material {
     // int8_t deterioration from water
 };
 
+struct MaterialColor {
+    const Material& material;
+    ColorId color;
+
+    inline MaterialColor(const MaterialColor& other) : material(other.material), color(other.color){
+    }
+
+
+    inline MaterialColor(MaterialColor&& other) : material(other.material), color(other.color){
+    }
+
+    inline MaterialColor(const Material& mat, ColorId col) : material(mat), color(col) {}
```

Comment:
Structs don't need these they are auto generated

Reply:
I don't know but this breaks things.

Reply:
You have to change the call to the ctor, left a comment specifying how